### PR TITLE
CAMS-283: Get Court Division Details

### PR DESCRIPTION
**Pull request by jamesobrooks**
# Purpose

Many places where we were getting office details were actually needing court division details.

# Major Changes

- Replace old OfficeDetails with CourtDivision.
- Add new courts endpoint to get CourtDivisionDetails where USTP office information was being retrieved but we really needed court division.

# Testing/Validation

Automated tests and builds pass.

### CODE REVIEW

File: backend/functions/courts/courts.function.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  beforeEach(() => {
+    request = createMockAzureFunctionRequest();
+    context = createMockAzureFunctionContext();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  jest
```
Comment (btposey) [inline backend/functions/courts/courts.function.test.ts:28]: Not necessary. ?

File: backend/functions/courts/courts.function.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    jest.spyOn(CourtsController.prototype, 'handleRequest').mockResolvedValue(camsHttpResponse);
+
+    const response = await handler(request, context);
+
+    expect(response).toEqual(azureHttpResponse);
+  });
+
+  test('should set error response', async () => {
+    const error = new CamsError('MOCK_OFFICES_CONTROLLER', {
```
Comment (btposey) [inline backend/functions/courts/courts.function.test.ts:49]: OFFICES -> COURTS or something generic.

File: backend/functions/lib/controllers/courts/courts.controller.test.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    const controller = new OfficesController();
+    await expect(async () => {
+      const camsHttpRequest = mockCamsHttpRequest();
+      applicationContext.request = camsHttpRequest;
+      await controller.handleRequest(applicationContext);
+    }).rejects.toThrow('Unknown error');
+    expect(getOfficeAttorneys).not.toHaveBeenCalled();
+  });
+
+  test('should call getOfficeAttorneys', async () => {
```
Comment (btposey) [inline backend/functions/lib/controllers/courts/courts.controller.test.ts:96]: Delete test

File: backend/functions/lib/controllers/courts/courts.controller.test.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    expect(attorneys).toEqual(
+      expect.objectContaining({
+        body: { meta: expect.objectContaining({ self: expect.any(String) }), data: [] },
+      }),
+    );
+    expect(getOffices).not.toHaveBeenCalled();
+    expect(getOfficeAttorneys).toHaveBeenCalledWith(applicationContext, officeCode);
+  });
+
+  test('should throw error for unsupported subResource', async () => {
```
Comment (btposey) [inline backend/functions/lib/controllers/courts/courts.controller.test.ts:118]: Delete test

File: backend/functions/lib/controllers/courts/courts.controller.test.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      return {
+        getOffices,
+        getOfficeAttorneys,
+        syncOfficeStaff,
+      };
+    }),
+  };
+});
+
+describe('offices controller tests', () => {
```
Comment (btposey) [inline backend/functions/lib/controllers/courts/courts.controller.test.ts:25]: This suite is testing offices controller not courts controller.

**Review by btposey — CHANGES_REQUESTED** [review]

File: backend/functions/courts/courts.function.test.ts (hunk 1 above)
Comment (jamesobrooks) [reply backend/functions/courts/courts.function.test.ts]: Seems so. Removed.

**Review by btposey — APPROVED** [review]

**Pull request merged**